### ISSUE TITLE
Tkinter app on macOS 10.14.6 causes WindowServer event port death

### ISSUE THREAD

**Issue by peterruhl** [pyinstaller/pyinstaller#4334]
**OS: macOS 10.14.6
Python: 3.6.6, 3.6.8, 3.7.4
Tkinter: 8.6 (bundled with python)
Pyinstaller: 3.5**

On macOS 10.14.6, when trying to run any python application using Tkinter that has been frozen, my WindowServer crashes, and brings me back to the macOS login screen. 
No issues while following the same procedure on macOS 10.14.5.

The minimum code to reproduce (found below) runs fine on its own on macOS 10.14.6

I have tried both methods found here: #3753  to ensure that TKinter is included.

I tried a number of different commands to freeze my application, all producing the same crash. Most recent command was `pyinstaller --windowed myapp.py` as suggested in issue #3753 

Output:

```
2019-07-25 09:30:21.328 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.328 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.328 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.328 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.329 Main[3268:17227] CGSTrackingRegionSetIsEnabled returned CG error 268435459
2019-07-25 09:30:21.342 Main[3268:17227] HIToolbox: received notification of WindowServer event port death.
2019-07-25 09:30:21.342 Main[3268:17227] port matched the WindowServer port created in BindCGSToRunLoop
```

Minimum code required to reproduce:

```

from tkinter import *
class Main:
    def __init__(self, parent):

        #Configure the main window
        self.myparent = parent
        self.myparent.wm_title("Please Don't Crash")
        self.myparent.geometry('440x300')
        
root = Tk()
myapp = Main(root)
root.mainloop()

```

This issue was reported on another project that uses pyinstaller as well. https://github.com/nh-server/fusee-interfacee-tk/issues/1

**Comment by pklein15** [comment]
MacOS 10.14.6 caused my 2018 MacBook Air to restart running Tkinter python script program. Worked fine before the update, then went to hell.

**Comment by tazzben** [comment]
I'm wondering if those reporting this issue have (or have had) Anaconda installed.  Note: I'm not saying you are executing the program with Anaconda, simply having it installed causes the problem.  See:

https://github.com/ContinuumIO/anaconda-issues/issues/11165

**Comment by ihaveamac** [comment]
I've never used anaconda, and still have this issue. In addition, I've noticed that the crash doesn't occur in a virtual machine (VMware), or in safe mode.

Also, I couldn't seem to reproduce the crash on the latest 10.15 Beta 5 (19A526h). [It was happening before.](https://github.com/nh-server/fusee-interfacee-tk/issues/3) Both times were a clean system (just installed, doesn't have Python 3.x otherwise)

**Comment by pklein15** [comment]
I do have anaconda installed, and it crashed both running it using an anaconda notebook and then from just a straight python (3.7.4 I think) prompt. I purchased the Mac Air on Friday and it worked fine. The automatic update installed Friday night. Failed Saturday of course.

**Comment by Achim63** [comment]
Same issue with an app build with py2app. I get kicked to the login screen and it worked fine in 10.14.5. I tried to rebuild it with Python 3.6.8 to no avail (now I'm back to 3.7.4).

**Comment by manish9909** [comment]
I have the same problem, have updated my mac and tkinter crashes my mac and logs me out.

**Comment by shuntera** [comment]
Same issue for me too on Mojave 10.14.6 and Python 3.7.3

**Comment by peterruhl** [comment]
@tazzben I've also never installed Anaconda on this machine.

**Comment by bentkidder** [comment]
Same issue for me on Mojave 10.14.6 and Python 3.7.3 running Anaconda.

**Comment by thuwzr** [comment]
Same issue for me on Mojave 10.14.6 and Python 3.73 runing pyplot.show

**Comment by zacek100** [comment]
Same here Mojave 10.14.6 and Python 3.72 runing pyinstaller 3.5

**Comment by sharahamo** [comment]
Same issue here. DO we have a fix yet?

**Comment by AnthonyCheetham** [comment]
Not a fix, but people googling the error message might come across this and find it useful. If you don't mind changing your matplotlib backend until this is fixed you can add a line to the start of your code to change to the MacOSX backend (or Qt4Agg etc.):
```
import matplotlib
matplotlib.use('MacOSX')
```

Or you can change this permanently by adding
`backend: MacOSX`
to your matplotlibrc file (probably in $HOME/.matplotlib/matplotlibrc )

**Comment by zacek100** [comment]
> Not a fix, but people googling the error message might come across this and find it useful. If you don't mind changing your matplotlib backend until this is fixed you can add a line to the start of your code to change to the MacOSX backend (or Qt4Agg etc.):
> 
> ```
> import matplotlib
> matplotlib.use('MacOSX')
> ```
> 
> Or you can change this permanently by adding
> `backend: MacOSX`
> to your matplotlibrc file (probably in $HOME/.matplotlib/matplotlibrc )

This did not work for me, same crash & error.

**Comment by tede12** [comment]
Same issue for me on Mojave 10.14.6 and Python 3.73 runing tkinter

this didn't solve the problem for me.
```python3
import matplotlib
matplotlib.use('MacOSX')
```

**Comment by sharahamo** [comment]
Hi all, the quick fix for now is just to downgrade python to 3.70. It works for me

**Comment by bentkidder** [comment]
Yes downgrading python is unfortunately the best option until there is a patch. Changing the matplotlib backend will not fix the issue for any other software employing tk

**Comment by zacek100** [comment]
Unfortunately downgrading to Python 3.7.0 did not work for me. I even tried with Python 3.6.8 and got same result. I think the issue might be with Tkinter 8.6.

**Comment by peterruhl** [comment]
I haven't tried 3.7.0 yet, but it was not working on 3.7.4, 3.6.8, or 3.6.6 for me when I reported this issue.

**Comment by Wayfear** [comment]
The same problem happened when I run an application that has a tkinter GUI in Python 2.7.16, but the problem solved when I run the application in Python 2.7.14

**Comment by newTypeGeek** [comment]
Same problem here. 
macOS 10.14.16
Python 3.7.3
Tkinter 8.6

**Comment by charkour** [comment]
Same problem here. I get kicked to the Apple login screen when trying to run a python script from the command line. This problem is new and has only happened after installing Anaconda.

**Comment by kemalty** [comment]
I also have the same problem.
MacOS Mojave 10.14.6 (18G95)

**Comment by Shanmugapriya03** [comment]
Facing the same issues as mentioned above Any workaround until it is fixed? @bjones1 @codewarrior0 @htgoebel @matysek @rasky

**Comment by toshreyansjain** [comment]
I am really stuck with this issue not able to do any python graphics on version 3.4

**Comment by nobumikiokada** [comment]
I experienced same issue on MacOS 10.14.6 and Python 3.7, 3.6, 2.7, please let me know if you have any workaround.

**Comment by maxime915** [comment]
Experienced same issue (Mac OS 10.14.6, Python 3.6.9, Tkinter 8.6, using anaconda) can report the crash come from the line `self.tk = _tkinter.create(screenName, baseName, className, interactive, wantobjects, useTk, sync, use)` in `$HOME/anaconda3/envs/myenv/lib/python3.6/tkinter/__init__.py` (l 2025) but the method called is defined in `$HOME/anaconda3/envs/myenv/lib/python3.6/lib-dynload/_tkinter.cpython-36m-darwin.so` which isn't user-readable nor editable.

I tried with another Mac (OS X 10.9.5, Python 3.7.4, Tinker 8.6, never touched anaconda): I'm unable reproduce the issue and to install anaconda properly on this computer so I can't check

**Comment by nobumikiokada** [comment]
> Experienced same issue (Mac OS 10.14.6, Python 3.6.9, Tkinter 8.6, using anaconda) can report the crash come from the line `self.tk = _tkinter.create(screenName, baseName, className, interactive, wantobjects, useTk, sync, use)` in `$HOME/anaconda3/envs/myenv/lib/python3.6/tkinter/__init__.py` (l 2025) but the method called is defined in `$HOME/anaconda3/envs/myenv/lib/python3.6/lib-dynload/_tkinter.cpython-36m-darwin.so` which isn't user-readable nor editable.
> 
> I tried with another Mac (OS X 10.9.5, Python 3.7.4, Tinker 8.6, never touched anaconda): I'm unable reproduce the issue and to install anaconda properly on this computer so I can't check

Thanks, I'm going to change previous MacOS in my Mac for workaround.

**Comment by nicozanf** [comment]
I've got the same problem with the frozen web2py framework on my repository. It uses py 3.7.4 and tk, but not Anaconda. It works fine before  Mac OS 10.14.6  ;-(

I confirm that the crash doesn't occur in a virtual machine (Virtualbox).

Nico

**Comment by begoechavarren** [comment]
Same problem here on a MacBook Pro 10.14.6 and Python 3.7.3

**Comment by nicozanf** [comment]
Hi,
I've found that in my case the problem is related to downloading the App from GitHub. It can be resolved (at least on Catalina) with any of the following workarounds:
- 'control' + click on downloaded file and then 'open' it (confirm the warnings). Then move the file in Applications and run it from there.
- download the files from another OS (Windows or Linux) and then use an USB key / drive in order to copy them to the Mac. This will not insert the security Extended Attribute onto them.
- after dowloading the app, on a Terminal go to that folder and run 'xattr -d com.apple.quarantine myapp.app' in order to delete the security Extended Attribute

Hope it helps you.

**Comment by podsrinivasa** [comment]
uninstalling Anaconda does the magic!!

**Comment by c0nb4** [comment]
Is it fixed in Catalina?

**Comment by newTypeGeek** [comment]
> Is it fixed in Catalina?
Yes, it is fixed.

**OS**: macOS Catalina 10.15.1
**Python**: 3.7.3 (Anaconda)
**Tkinter**: 8.6

**Comment by nobumikiokada** [comment]
Confirmed the issue was fixed in Catalina 10.15.1, thank you for everyone :)

**Comment by htgoebel** [comment]
Sounds like this was never relaed to PyInstaller. Closing.

**Comment by HappyLittleFish** [comment]
when I uninstall anaconda, and reinstall pyinstaller by  "brew install pyinstaller", it works . 
after this I check python version is not 3.7.6, back to 2.7.10. it works, never lagout.

**Comment by petioptrv** [comment]
I can confirm that this happens with Python 3.7.x. Moving to 3.8.3 fixes the problem on MacOS 10.14.6.

**Comment by vickywu1022** [comment]
It's possible to survive with Anaconda. Create a Conda environment with python 3.7 on MACOS 10.14.6 can help.
